lib/questionlib.php

```diff
@@ -687,26 +736,35 @@ function question_move_questions_to_category($questionids, $newcategoryid) {
             // mistakenly moved then the idnumber will not be completely lost.
             $q = new stdClass();
             $q->id = $question->id;
-            $q->category = $newcategoryid;
             $q->idnumber = $question->idnumber . '_' . $unique;
             $DB->update_record('question', $q);
         }
 
+        // Update the reference to point to the new category context.
+        $references = $DB->get_records('question_references',
```

Comment:
we didnt add a reference for each question in the reference table yet, that means it doesn't matter for the reference table when a question is moved to a new category, the references here are pointing to quiz, changing this means changing the quiz contexts to the new category context which breaks the data.

Reply:
code commented as discussed.